cdap-examples/LogAnalysis/src/main/java/co/cask/cdap/examples/loganalysis/LogAnalysisApp.java

```diff
@@ -0,0 +1,165 @@
+import co.cask.cdap.api.workflow.Workflow;
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import java.net.HttpURLConnection;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Application that demonstrate running a Spark and MapReduce simultaneously in a {@link Workflow}
+ */
+public class LogAnalysisApp extends AbstractApplication {
+
+  public static final String LOG_ANALYSIS_APP_NAME = "LogAnalysisApp";
+  public static final String LOG_STREAM = "logStream";
+  public static final String HIT_COUNTER_SERVICE = "HitCounterService";
+  public static final String RESPONSE_COUNTER_SERVICE = "ResponseCounterService";
+  public static final String RESPONSE_COUNT_STORE = "responseCount";
+  public static final String HIT_COUNT_STORE = "hitCount";
+
+  @Override
+  public void configure() {
+    setName(LOG_ANALYSIS_APP_NAME);
+    setDescription("CDAP Log Analysis App");
+
+    // A stream to ingest log data
+    addStream(new Stream(LOG_STREAM));
+
+    // A Spark and MapReduce for processing log data
+    addSpark(new SparkKMeansSpecification());
+    addMapReduce(new HitCounterProgram());
+
+    addWorkflow(new LogAnalysisWorkflow());
+
+    // Services to query for result
+    addService(HIT_COUNTER_SERVICE, new HitCounterServiceHandler());
+    addService(RESPONSE_COUNTER_SERVICE, new ResponseCounterHandler());
+
+    // Datasets to store output after processing
+    createDataset(RESPONSE_COUNT_STORE, KeyValueTable.class);
+    createDataset(HIT_COUNT_STORE, KeyValueTable.class);
+  }
+
+  /**
+   * A Workflow which ties spark and mapreduce program together for log analysis
+   */
+  public static class LogAnalysisWorkflow extends AbstractWorkflow {
+
+    @Override
+    public void configure() {
+      setName(LogAnalysisWorkflow.class.getSimpleName());
+      setDescription("Runs log analysis spark and mapreduce programs simultaneously");
+      fork()
+        .addMapReduce(HitCounterProgram.class.getSimpleName())
+        .also()
+        .addSpark(ResponseCounterProgram.class.getSimpleName())
+        .join();
+    }
+  }
+
+  /**
+   * Specification for the Spark program in this application
+   */
+  public static final class SparkKMeansSpecification extends AbstractSpark {
+
+    @Override
+    public void configure() {
+      setName(ResponseCounterProgram.class.getSimpleName());
+      setDescription("Counts the total number of responses for every unique response code");
+      setMainClass(ResponseCounterProgram.class);
+    }
+  }
+
+  /**
+   * A {@link Service} that responds with total number of hits for a given URL or path
+   */
+  public static final class HitCounterServiceHandler extends AbstractHttpServiceHandler {
+
+    private static final Gson GSON = new Gson();
+    public static final String URL_KEY = "url";
+    public static final String HIT_COUNTER_SERVICE_PATH = "hitcount";
```

Comment:
Shouldn't be public. If it is just for unit-test purpose, making it package private and annotate with `@VisibleForTesting`.